src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1287,6 +1287,19 @@ void IVisualStudioWorkspaceHost.OnAdditionalDocumentTextUpdatedOnDisk(DocumentId
                 _workspace.OnAdditionalDocumentTextUpdatedOnDisk(id);
             }
 
+            void IVisualStudioWorkspaceHost.OnIntellisenseBuildResult(ProjectId projectId, bool succeed)
+            {
+                // this can't be null
+                var state = _workspace.CurrentSolution.GetProjectState(projectId);
+                Contract.ThrowIfNull(state);
+
+                // if state is different than what we have
+                if (succeed != state.ProjectInfo.IsComplete)
+                {
+                    _workspace.OnProjectReloaded(state.ProjectInfo.WithIsComplete(succeed));
```

Comment:
Doesn't OnProjectReloaded mean we'll toss lots of caches and such? Shouldn't we have an OnProjectIsCompletedChanged or something to follow the pattern for everything else?